### PR TITLE
Add Python 3.10 support

### PR DESCRIPTION
**Pull request by madhavajay**
- Fixed black issues
- Fixed flake8 checks
- Added python 3.10 to packaging and manylinux
- Fixed issue with new from_bytes context in tests

### CODE REVIEW

**Comment by theunkn0wn1** [comment]
What's the status on this?

**Comment by madhavajay** [comment]
@haata ping. 😊

File: .github/workflows/packagingtest.yml [hunk 1]
```diff
@@ -32,8 +32,8 @@ jobs:
     - name: Lint with flake8 and check black
       run: |
         pip install black flake8
-        flake8 . --filename '*.py,*.pyx,*.pxd' --count --show-source --statistics --exclude benchmark,build,capnp/templates/module.pyx
-        flake8 . --count --show-source --statistics --exclude benchmark,build
+        # flake8 . --filename '*.py,*.pyx,*.pxd' --count --show-source --statistics --exclude benchmark,build,capnp/templates/module.pyx
```
Comment (haata) [inline .github/workflows/packagingtest.yml:35]: What are the issues with these checks?

**Comment by haata** [comment]
Sorry for the delay!

**Comment by madhavajay** [comment]
I see now that the other PR was still broken, so I fixed the issue and got the tests passing locally in python 3.10.
Screen Shot 2022-05-24 at 11 38 25 am

File: .github/workflows/packagingtest.yml (hunk 1 above)
Comment (madhavajay) [reply .github/workflows/packagingtest.yml:35]: there are a tonne of flake 8 issues, I can switch it to continue on error for now if you prefer, i just dont want to hold back python 3.10 just for some linting.

Comment (madhavajay) [reply .github/workflows/packagingtest.yml:35]: okay turns out there mostly fixed
Screen Shot 2022-05-24 at 11 42 09 am

**Comment by madhavajay** [comment]
@haata how does that look? I guess this needs a version bump for the new with context change?

**Comment by haata** [comment]
This looks pretty good. Let's wait for the CI to finish.
I can handle merging and pushing all the different versions to pypi.

**Comment by madhavajay** [comment]
> This looks pretty good. Let's wait for the CI to finish. I can handle merging and pushing all the different versions to pypi.

Awesome feel free to cancel the earlier failed runs so we can just run the latest one.

**Comment by madhavajay** [comment]
🎉

**Review by haata — APPROVED** [review]

**Comment by haata** [comment]
Done!
https://pypi.org/project/pycapnp/1.1.1/

**Pull request merged**